src/Core/Services/Implementations/SendService.cs

```diff
@@ -278,17 +281,17 @@ private async Task ValidateUserCanSaveAsync(Guid? userId, Send send)
             return;
         }
 
-        var disableSendPolicyCount = await _policyRepository.GetCountByTypeApplicableToUserIdAsync(userId.Value,
+        var disableSendPolicies = await _policyService.GetPoliciesApplicableToUserAsync(userId.Value,
             PolicyType.DisableSend);
-        if (disableSendPolicyCount > 0)
+        if (disableSendPolicies.Any())
         {
             throw new BadRequestException("Due to an Enterprise Policy, you are only able to delete an existing Send.");
         }
 
         if (send.HideEmail.GetValueOrDefault())
         {
-            var sendOptionsPolicies = await _policyRepository.GetManyByTypeApplicableToUserIdAsync(userId.Value, PolicyType.SendOptions);
-            if (sendOptionsPolicies.Any(p => p.GetDataModel<SendOptionsPolicyData>()?.DisableHideEmail ?? false))
+            var sendOptionsPolicies = await _policyService.GetPoliciesApplicableToUserAsync(userId.Value, PolicyType.SendOptions);
+            if (sendOptionsPolicies.Any(p => CoreHelpers.LoadClassFromJsonData<SendOptionsPolicyData>(p.PolicyData)?.DisableHideEmail ?? false))
```

Comment:
Is there any reason to use `CoreHelpers` here instead of the method provided by the `Policy` object? I assume that the more specific helper method (`Policy.GetDataModel`) is better to use because it encapsulates this behaviour for this specific object.

Reply:
oh yeah, of course 😁  that's fine